cordapp/src/integrationTest/kotlin/com/template/DriverBasedTest.kt

```diff
@@ -3,58 +3,70 @@ package com.template
-            // flows, and verifying the states in the vault and other important metrics to ensure that your CorDapp is
-            // working as intended.
-            assertEquals(partyAHandle.rpc.wellKnownPartyFromX500Name(bankB.name)!!.name, bankB.name)
-            assertEquals(partyBHandle.rpc.wellKnownPartyFromX500Name(bankA.name)!!.name, bankA.name)
-        }
+    fun `node test`() = withDriver {
+        // Start a pair of nodes and wait for them both to be ready.
+        val (partyAHandle, partyBHandle) = startNodes(bankA, bankB)
+
+        // From each node, make an RPC call to retrieve another node's name from the network map, to verify that the
+        // nodes have started and can communicate.
+
+        // This is a very basic test: in practice tests would be starting flows, and verifying the states in the vault
+        // and other important metrics to ensure that your CorDapp is working as intended.
+        assertEquals(bankB.name, partyAHandle.resolveName(bankB.name))
+        assertEquals(bankA.name, partyBHandle.resolveName(bankA.name))
     }
 
     @Test
-    fun `node webserver test`() {
-        driver(DriverParameters(isDebug = true, startNodesInProcess = true)) {
-            val nodeHandles = listOf(
-                    startNode(providedName = bankA.name),
-                    startNode(providedName = bankB.name)
-            ).map { it.getOrThrow() }
-
-            // This test starts each node's webserver and makes an HTTP call to retrieve the body of a GET endpoint on
-            // the node's webserver, to verify that the nodes' webservers have started and have loaded the API.
-            nodeHandles.forEach { nodeHandle ->
-                val webserverHandle = startWebserver(nodeHandle).getOrThrow()
-
-                val nodeAddress = webserverHandle.listenAddress
-                val url = "http://$nodeAddress/api/template/templateGetEndpoint"
-
-                val request = Request.Builder().url(url).build()
-                val client = OkHttpClient()
-                val response = client.newCall(request).execute()
-
-                assertEquals("Template GET endpoint.", response.body().string())
-            }
+    fun `node webserver test`() = withDriver {
+        // This test starts each node's webserver and makes an HTTP call to retrieve the body of a GET endpoint on
+        // the node's webserver, to verify that the nodes' webservers have started and have loaded the API.
+        startWebServers(bankA, bankB).forEach {
+            val request = Request.Builder()
+                .url("http://${it.listenAddress}/api/template/templateGetEndpoint")
+                .build()
+            val response = OkHttpClient().newCall(request).execute()
+
+            assertEquals("Template GET endpoint.", response.body().string())
         }
     }
+
+    // *********************
+    // * Utility functions *
```

Comment:
I like to use comments of the form `//region Utility functions` and then at the end `//endregion` because that way IntelliJ will fold the section automatically.